src/Diagnostics/DiagnosticsMachine/pointwise.jl

```diff
@@ -0,0 +1,140 @@
+"""
+    PointwiseDiagnostic
+
+A diagnostic with the same dimensions as the original grid (DG or
+interpolated). Mostly used to directly copy out prognostic or
+auxiliary state variables.
+"""
+abstract type PointwiseDiagnostic <: DiagnosticVar end
+function dv_PointwiseDiagnostic end
```

Comment:
Can / should we change this to `dv_pointwise_diagnostic`? (and similar with `dv_HorizontalAverage` etc.? I guess they're relatively internal methods, so not a hard blocker. (Also, I see that it nicely hints that the method is specifically defined for `PointwiseDiagnostic`). Maybe it's fine, actually.

Reply:
Also makes it easier to generate the function name: `dv_$(dvtype)`.